src/lib/netlist/plib/ptypes.h

```diff
@@ -177,9 +178,12 @@ namespace plib
 	#elif defined(__GLIBCXX__)
 		using cpp_stdlib = std::integral_constant<ci_cpp_stdlib, ci_cpp_stdlib::LIBSTDCXX>;
 		using cpp_stdlib_version = typed_version<(_GLIBCXX_RELEASE), 0>;
+	#elif defined(_MSC_VER)
+		using cpp_stdlib = std::integral_constant<ci_cpp_stdlib, ci_cpp_stdlib::MSC>;
+		using cpp_stdlib_version = typed_version<_MSC_VER / 100, _MSC_VER % 100>;
```

Comment:
This won’t actually work – `_MSC_VER` will be defined if you’re using MSVC itself or clang in MSVC ABI mode irrespective of the C++ standard library as it’s predefined by the compiler even when there’s no input.  It only tells you the version of the compiler int the case of MSVC itself, or for clang it’s an approximate MSVC version for feature parity (kind of like how clang defines `__GNUC__`, `__GNUC_MINOR__` and `__GNUC_PATCHLEVEL__` by default despite not being GCC).

As far as I can tell, there are no officially documented macros for detecting MSVCPRT or its version.  If you look at a simple header, say `<cstdint>` in different C++ standard libraries, you can see this.  The GNU libstdc++ version includes `<bits/c++config.h>` which contains the version and date macros, as well as a bunch of other stuff.  The libc++ version includes `<__config>` which serves the same purpose.  For MSVCPRT, the equivalent is `<yvals_core.h>`, and the version macros are `_CPPLIB_VER`, `_MSVC_STL_VERSION`, and `_MSVC_STL_UPDATE` (for me, they’re `650`, `142`, and `202104L`, resepctively).

If you look at Boost.C++ library source, they use the value of `_CPPLIB_VER` to detect MSVCPRT and determine which features/bugs should be expected.  It seems to be a single, monotonic integer value, not a major/minor version.  It’s probably best to just treat the value of `_CPPLIB_VER` as the major version of the library and leave the minor version and patch level as zero.  (And this is a place where major versions exceed one hundred by quite a bit.)